### DISCUSSION TITLE
Trigger car status update explicitly

### DISCUSSION THREAD

**Discussion by prog1407** [tobiasehlert/teslamateapi#114] [Q&A]
Hello everyone!
Thanks a lot for doing TeslaMateApi, it simplifies life for sure :) I'm using it right now in smart home automation scenarios to send different commands to my vehicle.

I would like to check if there is a way to explicitly force teslamate to check car status.
Right now I start the car at home, start driving and see that sometimes gps tracking begins only in 100-200 meters from home.
I assume that it is due to a fact that while car is not driving (and streaming is not in place) -- teslamate is polling for updates once in some period (30 seconds?). I guess in my case teslamate misses the real moment when car becomes online and I start driving. So it detects it a bit later due to a polling period time lag.

For me it is quite important to recognize that car is online and get gps data from very beginning of the ride. Because I use that gps data in my smart home automation for closing garage gate once the car is outside.

So my idea is to catch car online action with Tasker by detecting bluetooth connection between phone and Tesla. Tasker then can send some API request and notify teslamate that car status needs to be polled and updated asap.
This project does API calling of teslamate, is there any way to call some API that will also force data update inside of teslamate and initialize streaming connection?

Thanks a lot in advance!

**Comment by LelandSindt** [comment]
@prog1407  have a look at this discussion. 

https://github.com/adriankumpf/teslamate/discussions/1433

**Reply by prog1407** [reply]
thanks a lot, that is very helpful! I've added calling of the specified method. However as I see it doesn't force TeslaMate to do the update immediately but just schedules the new one for 30 seconds or so. 
Basically what I did right now with my TeslaMate+TeslaMateApi+Openhab(MQTT). I have a switch that enables climate in my Tesla  (through TeslaMateApi). But TeslaMate becomes aware about that change only in 30 seconds or so while it would be helpful to get that update sooner.  
So it appears that there is no way to explicitly trigger the update in TeslaMate (or change pooling interval to custom for some time) while it looks like a cool feature to me :)
